Base4/questions.js

```diff
@@ -1,101 +1,126 @@
-
-var stringSize = function (text) {
-
+var stringSize = function(text) {
+  return text.length;
 }
-var replaceCharacterE = function (text) {
-
+var replaceCharacterE = function(text) {
+  return text.replace("e", " ");
 }
-var concatString = function (text1, text2) {
-
+var concatString = function(text1, text2) {
+  return text1 + text2;
 }
-var showChar5 = function (text) {
-
+var showChar5 = function(text) {
+  return text.charAt(4);
 }
-var showChar9 = function (text) {
-
+var showChar9 = function(text) {
+  return text.slice(0, 9);
 }
-var toCapitals = function (text) {
-
+var toCapitals = function(text) {
+  return text.toUpperCase();
 }
-var toLowerCase = function (text) {
-
+var toLowerCase = function(text) {
+  return text.toLowerCase();
 }
-var removeSpaces = function (text) {
-
+var removeSpaces = function(text) {
+  return text.trim();
 }
-var IsString = function (text) {
-
+var IsString = function(text) {
+  return (typeof(text) == "string");
 }
 
-var getExtension = function (text) {
-
+var getExtension = function(text) {
+  return text.slice(text.lastIndexOf(".") + 1, text.length);
 }
-var countSpaces = function (text) {
-
+var countSpaces = function(text) {
+  let counter = 0;
+  for (var i = 0; i < text.length; i++) {
+    if (text.charAt(i) == " ") counter++;
+  }
+  return counter;
 }
-var InverseString = function (text) {
+var InverseString = function(text) {
+  return text.split("").reverse().join("");
 
 }
 
-var power = function (x, y) {
-
+var power = function(x, y) {
+return Math.pow(x,y);
 }
-var absoluteValue = function (num) {
-
+var absoluteValue = function(num) {
+if (num<0)return -num;
+return num;
 }
-var absoluteValueArray = function (array) {
-
+var absoluteValueArray = function(array) {
+var absoluteArray = new Array();
```

Comment:
If you're gonna use `new Array`, may as well specify it's size:
```js
var absoluteArray = new Array(array.length);
```

But moreover, you could re-use the previous function, which has been, presumably, tested:

```js
for(var i=0;i<array.length;i++){
  absoluteArray[i] = absoluteValue(array[i])
}
```

Or better, use `map` :)